client/src/pages/userview.js

```diff
@@ -28,7 +28,7 @@ const navigateToAddMeal = () => {
   navigate(`/api/nutrition`)
 }
 
-
+//unsure how to add id back in here need them for delete all need to to have as keys ??
 let meals = {};
 let totalCalories = 0;
 
```

Comment:
There are a bunch of ways you could do it, but here's one idea that might help:

```
usermeals.forEach((meal) => {
  meals[entry.mealcourse].push(meal);
  totalCalories += Number(meal.calories)
});
```

Reply:
Then on line 51 you could do something like
```
 Object.entries(meals.Breakfast).map((food) => {
          return (<li id={food.id}>{food.foodeaten} CAL:{food.calories}</li>);
        })

```

I would recommend using objects over lists so it is easier to read the code. When I read you code and saw `food[0]` or `food[1]` I had to guess what those items are, plus in the original map you are doing above, you are throwing the `id` away which you want for the `li` elements.